Dockerfile

```diff
@@ -289,6 +289,13 @@ RUN DART_ARCH=${TARGETARCH} \
   && rm "/tmp/${DART_EXE}" \
   && dart --version
 
+### DOTNET
+ENV DOTNET_NOLOGO=true \
+  DOTNET_CLI_TELEMETRY_OPTOUT=true
+
+RUN curl -sSL https://dot.net/v1/dotnet-install.sh | bash /dev/stdin --channel 6.0 --install-dir /usr/share/dotnet \
```

Comment:
I see both `6.0` and `7.0` listed on https://dotnet.microsoft.com/en-us/download

I'm totally ignorant of dotnet / nuget versioning expectations.

Did anything change from a manifest / lockfile perspective between them? Is `7.0` backwards compatible with `6.0` files? If so, should we use that? Or is it better to stick with `6.0`  since it's LTS?

Reply:
Also, is it typical to only pin to the latest minor version, or should we pin to the full patch release? Not sure how frequent/stable the patch releases tend to be...

Reply:
.NET versions are backwards compatible, so yes, `7.0` works with `6.0`. For updating packages and lockfiles the only universal option is to use the latest preview SDK version as older SDK versions cannot work with projects that specify newer SDK versions. Thus, this PR should not hardcode the `6.0` version as it will break when working with projects using `7.0+`. Granted I'm not super familiar with how dependabot manages toolchain versions.

Patch releases are released once a month. Most projects I've seen only pin to the latest minor version.

Relevant docs:

- [Overview of how .NET is versioned](https://learn.microsoft.com/en-us/dotnet/core/versions/)
- [Breaking changes in .NET](https://learn.microsoft.com/en-us/dotnet/core/compatibility/breaking-changes)

Reply:
Thank you, that's very helpful.

Typically for most ecosystems we pin the toolchain and then manually bump it as needed (ironic, I know 😁, we'll get :dependabot: there over time).

Pinning to minor rather than patch is probably fine, we can always get more specific if needed.

So we'd hardcode to `7.0` at the start, and bump to `7.1` or `8.0` once those get released.

Reply:
Maybe outside of the scope of this PR, but could dependabot read and use the highest .NET version specified in the project configuration? I believe Mend Renovate does something like this.

This would ensure dependabot uses the version used by the developer and avoids potential breaking changes due to using a newer version. It would also allow dependabot to work with preview SDK versions which at least I'd find useful.

Reply:
Thanks for the feedback, but definitely outside the scope of this PR. 

If an issue already exists for these, it's probably tagged with the `L: dotnet:nuget` label: https://github.com/dependabot/dependabot-core/issues?q=is%3Aopen+is%3Aissue+label%3A%22L%3A+dotnet%3Anuget%22

Otherwise please file new issues.